tests/openflow_test.go

```diff
@@ -8,81 +8,91 @@ import (
-			{"ovs-vsctl add-port br-test1 intf2 -- set interface intf2 type=internal", true},
-			{"ovs-ofctl add-flow br-test1 table=0,priority=0,in_port=1,actions=drop", true},
-		},
-
-		tearDownCmds: []helper.Cmd{
-			{"ovs-vsctl del-br br-test1", true},
-		},
-
-		check: func(c *TestContext) error {
-			gh := c.gh
-			gremlin := "g"
-			if !c.time.IsZero() {
-				gremlin += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
-			}
-
-			gremlin += `.V().Has("Type", "ovsbridge", "Name", "br-test1")`
-			gremlin += `.Out("Type", "ofrule").Has("actions","drop")`
-
-			nodes, err := gh.GetNodes(gremlin)
-			if err != nil {
-				return err
-			}
+type ruleCmd struct {
+	rule string
+	add  bool
+}
 
-			if len(nodes) != 1 {
-				return fmt.Errorf("Expected 1 node, got %+v", nodes)
-			}
+func verify(c *TestContext, expected []int) error {
+	for i, e := range expected {
+		gh := c.gh
+		gremlin := "g"
+		if !c.time.IsZero() {
+			gremlin += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
+		}
+
+		gremlin += `.V().Has("Type", "ovsbridge", "Name", "br-test1")`
+		gremlin += fmt.Sprintf(`.Out("Type", "ofrule").Has("actions","resubmit(;%d)")`, i+1)
+		nodes, err := gh.GetNodes(gremlin)
+		if err != nil {
+			return err
+		}
+		l := len(nodes)
+		if l != e {
+			return fmt.Errorf("expected %d rules with 'resubmit(,%d)' but got %d", e, i+1, l)
+		}
+	}
+	return nil
+}
 
-			return nil
-		},
+func makeTest(t *testing.T, rules []ruleCmd, expected []int) {
+	setupCmds := []helper.Cmd{
+		{"ovs-vsctl add-br br-test1", true},
+		{"ovs-vsctl add-port br-test1 intf1 -- set interface intf1 type=internal", true},
+		{"ovs-vsctl add-port br-test1 intf2 -- set interface intf2 type=internal", true},
 	}
 
-	RunTest(t, test)
-}
+	for _, ruleCmd := range rules {
+		var cmd string
+		if ruleCmd.add {
+			cmd = fmt.Sprintf("ovs-ofctl add-flow br-test1 %s", ruleCmd.rule)
+		} else {
+			cmd = fmt.Sprintf("ovs-ofctl del-flows --strict br-test1 %s", ruleCmd.rule)
+		}
+		setupCmds = append(setupCmds, helper.Cmd{Cmd: cmd, Check: true})
+	}
 
-func TestDelOFRule(t *testing.T) {
 	test := &Test{
-		setupCmds: []helper.Cmd{
-			{"ovs-vsctl add-br br-test1", true},
-			{"ovs-vsctl add-port br-test1 intf1 -- set interface intf1 type=internal", true},
-			{"ovs-vsctl add-port br-test1 intf2 -- set interface intf2 type=internal", true},
-			{"ovs-ofctl add-flow br-test1 table=0,priority=0,in_port=1,actions=drop", true},
-			{"ovs-ofctl del-flows br-test1 table=0,in_port=1", true},
-		},
+		setupCmds: setupCmds,
 
 		tearDownCmds: []helper.Cmd{
 			{"ovs-vsctl del-br br-test1", true},
 		},
 
 		check: func(c *TestContext) error {
-			gh := c.gh
-			gremlin := "g"
-			if !c.time.IsZero() {
-				gremlin += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
-			}
-
-			gremlin += `.V().Has("Type", "ovsbridge", "Name", "br-test1")`
-			gremlin += `.Out("Type", "ofrule").Has("actions","drop")`
+			return verify(c, []int{1})
```

Comment:
this line looks incorrect (debug stuff remaining ?), expected is never used

Reply:
Yes factorization of tests made them useless. TestDelRule was in fact even false. Because of the strict mode of ovs-ofctl del-flows required by the third test (in that case, the rule priority must be given), the rule was not removed.